account/manager.go

```diff
@@ -107,6 +108,65 @@ func (wt witnessType) witnessSize() (int, error) {
 	}
 }
 
+// Action is a type of account modification.
+type Action string
+
+const (
+	CREATE   Action = "create"
+	DEPOSIT  Action = "deposit"
+	WITHDRAW Action = "withdraw"
+	RENEW    Action = "renew"
+	CLOSE    Action = "close"
+)
+
+type TxLabel struct {
+	Account AccountTxLabel `json:"account"`
+}
+
+type AccountTxLabel struct {
+	Key           string         `json:"key"`
+	Action        Action         `json:"action"`
+	ExpiryHeight  uint32         `json:"expiry_height"`
+	OutputIndex   uint32         `json:"output_index"`
+	IsExpirySpend bool           `json:"expiry_spend"`
+	TxFee         btcutil.Amount `json:"tx_fee"`
+	BalanceDiff   btcutil.Amount `json:"balance_diff"`
+}
+
+// actionTxLabel returns a transaction label for use with account
+// modification actions.
+func actionTxLabel(account Account, action Action, isExpirySpend bool,
+	txFee btcutil.Amount, balanceDiff btcutil.Amount,
+) (*string, error) {
+
+	prefixTag := "poold -- "
+
+	acctKey := account.TraderKey.PubKey.SerializeCompressed()
+	label := AccountTxLabel{
+		Key:           string(acctKey),
+		Action:        action,
+		ExpiryHeight:  account.Expiry,
+		OutputIndex:   account.OutPoint.Index,
+		IsExpirySpend: isExpirySpend,
+		TxFee:         txFee,
+		BalanceDiff:   balanceDiff,
+	}
+	labelJson, err := json.Marshal(label)
+	if err != nil {
+		return nil, err
+	}
+
+	// Transaction labels are limited in size. Therefore, compact the JSON.
+	labelCompactJson := new(bytes.Buffer)
+	err = json.Compact(labelCompactJson, labelJson)
+	if err != nil {
+		return nil, err
+	}
+
+	labelString := prefixTag + labelCompactJson.String()
```

Comment:
Should use `fmt.Sprintf(prefixTag, string(labelJson))` here and add a `%s` to the `prefixTag` to avoid some gotchas around string conversion in Golang.

Reply:
I'm going to try `fmt.Sprintf(prefixTag, labelJson)` because I think the `%s` should take care of converting `labelJson` into a string.